File: exts/jphp-gui-game-ext/src/main/java/org/develnext/jphp/ext/game/classes/event/UXCollisionEvent.java
package org.develnext.jphp.ext.game.classes.event;

import javafx.event.EventTarget;
import org.develnext.jphp.ext.game.GameExtension;
import org.develnext.jphp.ext.game.support.CollisionEvent;
import org.develnext.jphp.ext.game.support.Vec2d;
import org.develnext.jphp.ext.javafx.classes.event.UXEvent;
import php.runtime.annotation.Reflection.Namespace;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Namespace(GameExtension.NS + "\\event")
public class UXCollisionEvent extends UXEvent {
    interface WrappedInterface {
        @Property Vec2d normal();
        @Property Vec2d tangent();
    }

    public UXCollisionEvent(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    public UXCollisionEvent(Environment env, CollisionEvent wrappedObject) {
        super(env, wrappedObject);
    }

    @Override
    public CollisionEvent getWrappedObject() {
        return (CollisionEvent) super.getWrappedObject();
    }

    @Signature
    public void __construct(CollisionEvent parent, Object sender, Object target) {
        __wrappedObject = new CollisionEvent(parent, sender, (EventTarget) target);
    }
}

File: exts/jphp-gui-game-ext/src/main/resources/JPHP-INF/sdk/php/game/event/UXCollisionEvent.php
<?php
namespace php\game\event;

use php\gui\event\UXEvent;

/**
 * ->sender first object
 * ->other second object
 *
 * Class UXCollisionEvent
 * @package php\game\event
 * @packages game, javafx
 */
class UXCollisionEvent extends UXEvent
{
    /**
     * @var float[]
     */
    public $normal = [0, 0];

    /**
     * @var float[]
     */
    public $tangent = [0, 0];

    /**
     * UXCollisionEvent constructor.
     * @param UXCollisionEvent $parent
     * @param mixed $sender
     * @param mixed $target
     */
    public function __construct(UXCollisionEvent $parent, $sender, $target)
    {
    }
}